src/main/java/com/mulberry/blackmulberry/service/UserService.java

```diff
@@ -0,0 +1,32 @@
+package com.mulberry.blackmulberry.service;
+
+import com.mulberry.blackmulberry.entity.User;
+import com.mulberry.blackmulberry.exception.DuplicateResourceException;
+import com.mulberry.blackmulberry.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public void save(User user) throws DuplicateResourceException {
+        if (userRepository.findByEmail(user.getEmail()).isPresent()){
+            throw new DuplicateResourceException("User already exists!");
+        }
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        userRepository.save(user);
+    }
+
+    public Optional<User> findByEmail(String email){
+        return userRepository.findByEmail(email);
```

Comment:
why you return Optional ?

Reply:
but the userRepository method returns me an Optional, can you send me how to do without the optional?